src/lib/ui/ProBeneficiaryCreate/beneficiary.schema.ts

```diff
@@ -1,26 +1,23 @@
-import { isDate } from 'date-fns';
+import { validateCodePostal, validateDateInput, validatePhoneNumber } from '$lib/validation';
 import * as yup from 'yup';
 
 export const beneficiaryAccountSchema = yup.object().shape({
 	firstname: yup.string().required(),
 	lastname: yup.string().required(),
-	dateOfBirth: yup.string().test('is-date-valid', 'Le format de la date est incorrect', (value) => {
-		if (!/[0-9]{4}-[0-9]{2}-[0-9]{2}/.test(value)) {
-			return false;
-		}
-		const date = new Date(value);
-		return isDate(date);
-	}),
+	dateOfBirth: yup
+		.string()
+		.test('is-date-valid', 'Le format de la date est incorrect', (value) => {
+			return validateDateInput(value);
+		})
+		.required(),
 
 	mobileNumber: yup
 		.string()
 		.test('is-phone-valid', 'Le format du téléphone est incorrect', (value) => {
-			/* TODO:
-			 * Apparently, test() seems to override nullable, so we currently accept empty
-			 * values manually, but this should be fixed cleanly later on.
-			 **/
-			const hotfix = value === '';
-			return hotfix || /^(?:(?:\+|00)33|0)\s*[1-9](?:[\s.-]*\d{2}){4}$/.test(value.trim());
+			if (value) {
+				return validatePhoneNumber(value);
+			}
+			return true;
```

Comment:
suggestion: mais peut-être moins lisible ?
```suggestion
  		return !value || validatePhoneNumber(value);
```

Reply:
oui c'est moins lisible :)